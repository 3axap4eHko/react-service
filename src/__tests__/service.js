import React, { Component, Children } from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import withService, { fetchServices, importData } from '../index';

configure({ adapter: new Adapter() });

beforeEach(() => {

});

test('withService HOC test', async () => {
  const counters = {
    service: 0,
    render: 0,
  };

  const service = () => ++counters.service && new Promise(r => setInterval(r, 1000, 'test'));

  @withService(service)
  class Data extends Component {
    render() {
      counters.render++;
      return <div>{JSON.stringify(this.props.data)}</div>;
    }
  }

  const app = (<Data />);

  const cache = await fetchServices(app);
  expect(counters.render).toBe(1);
  importData(cache);
  const component = renderer.create(app);
  expect(component.toJSON()).toMatchSnapshot();
  expect(counters.service).toBe(1);
  expect(counters.render).toBe(2);
  expect(Data.displayName).toBe('ServiceConnector-0-service(Data)');
});