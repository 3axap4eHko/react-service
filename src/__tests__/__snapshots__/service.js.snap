// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withService HOC test 1`] = `
<div>
  {"result":"test","loading":false,"error":null}
</div>
`;
